data/graph_db_utils.py

```diff
@@ -0,0 +1,44 @@
+import os
+from dotenv import load_dotenv
+from neo4j import GraphDatabase
+
+load_dotenv()
+
+URI = os.getenv("NEO4J_URI")
+AUTH = (os.getenv("NEO4J_USERNAME"), os.getenv("NEO4J_PASSWORD"))
+
+driver = GraphDatabase.driver(URI, auth=AUTH)
+
+def test_connection():
+    try:
+        driver.verify_connectivity()
+        print("database connection active")
+        
+    except Exception as e:
+        print(f"Error: {e}")
+        
+    finally:
+        driver.close()
+        
+
+
+def create_node(name, description):
+    try:
+        session = driver.session()
+        query = """
+        CREATE (g:Goal {name: $name, description: $description})
```

Comment:
I've read that `MERGE` will create nodes if they don't exist or take no action if the node already exists.

I'm not 100% sure on what best practice is but wondering if we want to use `MERGE` to avoid potentially creating duplicate nodes?